recipes/pexports/all/conanfile.py

```diff
@@ -1,94 +1,109 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-import contextlib
 import os
 
-required_conan_version = ">=1.33.0"
+from conan import ConanFile
+from conan.tools.build import cross_building
+from conan.tools.env import Environment, VirtualBuildEnv, VirtualRunEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, chdir, replace_in_file
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import unix_path, is_msvc
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.52.0"
 
 
 class PExportsConan(ConanFile):
     name = "pexports"
     description = "pexports is a program to extract exported symbols from a PE image (executable)."
-    homepage = "https://sourceforge.net/projects/mingw/files/MinGW/Extension/pexports/"
     license = "GPL-2.0-or-later"
-    topics = ("windows", "dll", "PE", "symbols", "import", "library")
     url = "https://github.com/conan-io/conan-center-index"
-    settings = "os", "arch", "compiler", "build_type"
-
-    exports_sources = "patches/*"
-
-    _autotools = None
+    homepage = "https://sourceforge.net/projects/mingw/files/MinGW/Extension/pexports/"
+    topics = ("windows", "dll", "PE", "symbols", "import", "library")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    package_type = "application"
+    settings = "os", "arch", "compiler", "build_type"
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
+    def export_sources(self):
+        export_conandata_patches(self)
+
     def configure(self):
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
 
-    def build_requirements(self):
-        self.build_requires("automake/1.16.3")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def package_id(self):
         del self.info.settings.compiler
 
-    def source(self):
-        filename = "pexports.tar.xz"
-        tools.get(**self.conan_data["sources"][self.version], filename=filename,
-                  destination=self._source_subfolder, strip_root=True)
+    def validate(self):
+        if self.settings.arch in ["armv8", "armv8.3"] and cross_building(self):
+            raise ConanInvalidConfiguration("pexports does not support cross-compilation to armv8")
```

Comment:
```suggestion
        # FIXME: Need to build with Mac M1
        if self.settings.arch in ["armv8", "armv8.3"] and cross_building(self):
            raise ConanInvalidConfiguration(f"Conan recipe {self.ref} does not support armv8. Contributions are welcome!")
```

Let's make it more clear, so, people can add contribution in the future.

I see no problem restricting a package in case there is a struggle to build, but need make it clear to ask contribution.